src/main/scala/vsys/blockchain/state/diffs/LeaseTransactionsDiff.scala

```diff
@@ -45,19 +40,11 @@ object LeaseTransactionsDiff {
       lease <- leaseEi
       recipient = lease.recipient
       isLeaseActive = s.isLeaseActive(lease)
-      leaseProofsHead <- lease.proofs.proofs.headOption match {
-        case Some(x) => Right(x)
-        case _ => Left(EmptyProofs)
-      }
-      leaseProof <- EllipticCurve25519Proof.fromBytes(leaseProofsHead.bytes.arr)
+      leaseProof <- lease.proofs.firstCurveProof
       leaseSender = leaseProof.publicKey
       _ <- if (!isLeaseActive)
```

Comment:
Just as what I talked in last pr. this check can be executed before leaseProof, ;-)

Reply:
OK, i've fixed this